### ISSUE TITLE
Theme question: only the 2022 version?

### ISSUE THREAD

**Issue by jason-mehmel** [garglk/garglk#686]
I've got the 2019 version from Debian Sid, and I can't find the themes listed in the themes.md file here... but I also don't see any mention of themes in the release info! (Though I may have missed it.)

**Comment by cspiegel** [comment]
Yes (to answer the subject), proper theme support won't be included till the next release. However, that next release is coming up soon (hopefully within a month), and if you'd rather not wait, the [Actions tab](https://github.com/garglk/garglk/actions) provides continuous builds for various platforms.  The current master should be considered stable, so there's not much risk in running it.

There's not a continuous Debian package, as Linux vendors/distributions wind up being responsible for packaging, but as a stop-gap, the AppImage should run on most Linux platforms.

**Comment by cspiegel** [comment]
To clarify a misstatement: for some reason I assumed you were talking about the upcoming 2022.2 release (as if you'd know about that), not the current 2022.1: theme support isn't in any _released_ version of Gargoyle yet at all.

**Comment by jason-mehmel** [comment]
Thank you for both replies!

Yeah, I think I'd assumed it was part of the current (released) 2022 version based on what I'd read elsewhere, though at this point I couldn't track down exactly where.

I'll give that more recent build / appimage a spin to check out themes on that version!

**Comment by jason-mehmel** [comment]
(Okay, after poking around in the actions tab, I'm not sure how to actually access an appimage download... mind if I get a newbie instruction on that?)

**Comment by cspiegel** [comment]
Yeah, it's definitely not an intuitive process.  In the Actions tab, navigate to the AppImage action on the left, then select the latest run on the right which corresponds to the "master" branch.  There should be an artifact at the bottom that's downloadable (as long as you're logged in to a Github account).

But that's all really annoying, so I'll provide a link here that uses https://nightly.link to ease access to artifacts: https://nightly.link/garglk/garglk/workflows/appimage/master/Gargoyle-x86_64.AppImage

This should always have the latest AppImage from the master branch.

**Comment by jason-mehmel** [comment]
Thank you for this!

I also deeply appreciate your patience and politeness in responding to these questions!

**Comment by jason-mehmel** [comment]
And one more note: I found the ability to switch themes inside the config file very easy!

Are the themes in the code folder here on github applicable if we add them to the appropriate local folders?

**Comment by cspiegel** [comment]
> Are the themes in the code folder here on github applicable if we add them to the appropriate local folders?

Yes, if you copy the entire `themes` directory to `$HOME/.local/share/io.github.garglk/Gargoyle` (assuming that's where Gargoyle's looking on your system, which is typical), then the themes will be available.

Now, the AppImage _should_ contain the themes, and in fact does, but Gargoyle can't find them there.  I'll try to figure out what's going on with that.  But for now, your proposed workaround should be sufficient.

AppImages use a bit of trickery to fool the program into thinking it's installed "normally" on the system, but for whatever reason it's failing with Gargoyle: a normal install does not require you to put the themes in your home directory, but will install them site-wide and have them available by default.  That's ultimately what should happen with the AppImage, too.
